packages/flutter_tools/lib/src/commands/build_aot.dart

```diff
@@ -100,8 +100,12 @@ String buildAotSnapshot(
   } else {
     String artifactsDir = tools.getEngineArtifactsDirectory(platform, buildMode).path;
     entryPointsDir = artifactsDir;
-    String hostToolsDir = path.join(artifactsDir, getNameForHostPlatform(getCurrentHostPlatform()));
-    genSnapshot = path.join(hostToolsDir, 'gen_snapshot');
+    if (platform == TargetPlatform.ios) {
```

Comment:
Yeah, this part got a little messy.  Once I get this working, I'll come back and refactor.